integration/test/Testlib/RunServices.hs

```diff
@@ -0,0 +1,135 @@
+{-# OPTIONS_GHC -Wno-unused-matches #-}
+
+module Testlib.RunServices where
+
+import Control.Monad.Codensity (lowerCodensity)
+import Control.Monad.Trans (MonadIO (liftIO))
+import qualified Data.Map as Map
+import System.Directory
+import System.Environment (getArgs)
+import System.Exit (exitWith)
+import System.FilePath
+import System.Posix (getWorkingDirectory)
+import System.Process (createProcess, proc, waitForProcess)
+import Testlib.ModService
+import Testlib.Prelude
+import Testlib.ResourcePool
+import Testlib.Run (createGlobalEnv)
+
+backendA :: BackendResource
+backendA =
+  BackendResource
+    { berBrigKeyspace = "brig_test",
+      berGalleyKeyspace = "galley_test",
+      berSparKeyspace = "spar_test",
+      berGundeckKeyspace = "gundeck_test",
+      berElasticsearchIndex = "directory_test",
+      berFederatorInternal = 8097,
+      berFederatorExternal = 8098,
+      berDomain = "example.com",
+      berAwsUserJournalQueue = "integration-user-events.fifo",
+      berAwsPrekeyTable = "integration-brig-prekeys",
+      berAwsS3Bucket = "dummy-bucket",
+      berAwsQueueName = "integration-gundeck-events",
+      berBrigInternalEvents = "integration-brig-events-internal",
+      berEmailSMSSesQueue = "integration-brig-events",
+      berEmailSMSEmailSender = "backend-integration@wire.com",
+      berGalleyJournal = "integration-team-events.fifo",
+      berVHost = "/",
+      berNginzSslPort = 8443
+    }
+
+staticPortsA :: Map.Map Service Word16
+staticPortsA =
+  Map.fromList
+    [ (Brig, 8082),
+      (Galley, 8085),
+      (Gundeck, 8086),
+      (Cannon, 8083),
+      (Cargohold, 8084),
+      (Spar, 8088),
+      (BackgroundWorker, 8089),
+      (Nginz, 8080)
+    ]
+
+backendB :: BackendResource
+backendB =
+  BackendResource
+    { berBrigKeyspace = "brig_test2",
+      berGalleyKeyspace = "galley_test2",
+      berSparKeyspace = "spar_test2",
+      berGundeckKeyspace = "gundeck_test2",
+      berElasticsearchIndex = "directory2_test",
+      berFederatorInternal = 9097,
+      berFederatorExternal = 9098,
+      berDomain = "b.example.com",
+      berAwsUserJournalQueue = "integration-user-events.fifo2",
+      berAwsPrekeyTable = "integration-brig-prekeys2",
+      berAwsS3Bucket = "dummy-bucket2",
+      berAwsQueueName = "integration-gundeck-events2",
+      berBrigInternalEvents = "integration-brig-events-internal2",
+      berEmailSMSSesQueue = "integration-brig-events2",
+      berEmailSMSEmailSender = "backend-integration2@wire.com",
+      berGalleyJournal = "integration-team-events.fifo2",
+      -- TODO
+      berVHost = "/",
```

Comment:
@battermann do you know what vhost backend B (b.example.com) should have?

Reply:
Not sure. AFAIK the rabbitmq queues are named after the target backend. So, messages going to `example.com` go into the queue named `example.com`. If multiple backends have the same target backend, this will cause a clash. But if there are only 2 backends (like in the old integration tests `example.com` and `b.example.com`), this doesn't matter. However, if backends `example.com` and `b.example.com` want to send messages to the dynamic backends, this can cause a problem. If we ever want integration tests where the old backends talk to the dynamic backends, we need different vhosts for the old backends. The convention is to name the vhost after the domain. I'd propose for consistency to create vhosts for backend A and B, named `example.com` and `b.example.com`. These can be created via curl:

```
curl -u "$RABBITMQ_USERNAME":"$RABBITMQ_PASSWORD" -X PUT "$ENDPOINT_URL"/example.com
```